pkg/network/namescheme/networknamescheme.go

```diff
@@ -92,14 +92,22 @@ func CreateOrdinalNetworkNameScheme(vmiNetworks []v1.Network) map[string]string
 	return networkNameSchemeMap
 }
 
+// OrdinalPodInterfaceName returns the ordinal interface name for the given network name.
+// networks contains pod networks and multus networks.
+// Primary network will use "eth0" and the secondary ones will use "net<id>" format, where id is an enumeration
+// from 1 to n.
 func OrdinalPodInterfaceName(name string, networks []v1.Network) string {
-	for i, network := range networks {
-		if network.Name == name {
-			if vmispec.IsSecondaryMultusNetwork(network) {
-				return generateOrdinalInterfaceName(i)
+	secondaryID := 0
+	for _, network := range networks {
+		if vmispec.IsSecondaryMultusNetwork(network) {
+			secondaryID++
+			if network.Name == name {
+				return generateOrdinalInterfaceName(secondaryID)
+			}
+		} else {
+			if network.Name == name {
```

Comment:
It seems correct to me.
But there are a lot of conditions here, making this a bit messy.

What do you think about this option:
```go
if primaryNetwork := vmispec.LookUpDefaultNetwork(networks); primaryNetwork != nil && primaryNetwork.Name == name {
    return PrimaryPodInterfaceName
}
for i, network := range FilterMultusNonDefaultNetworks(networks) {
    if network.Name == name {
        return generateOrdinalInterfaceName(i)
    } 
}
return ""
```

Reply:
Hi @EdDev,
Thanks for the tips!

Your suggestion makes more clarify to me.
I accepted your suggestion and forced push again.

Please help to recheck, Thanks!